src/main/scala/com/fasterxml/jackson/module/scala/modifiers/ScalaTypeModifier.scala

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.module.scala.modifiers
+
+import java.lang.reflect.{ParameterizedType, Type}
+
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.`type`._
+import com.fasterxml.jackson.module.scala.JacksonModule
+
+class ScalaTypeModifier extends TypeModifier {
+  override def modifyType(javaType: JavaType,
+                          jdkType: Type,
+                          context: TypeBindings,
+                          typeFactory: TypeFactory): JavaType = {
+
+
+    if (javaType.isTypeOrSubTypeOf(classOf[collection.Map[_, _]])) {
+      MapLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0), javaType.containedTypeOrUnknown(1))
```

Comment:
As with `ReferenceType` and `CollectionLikeType`, I think exact check for `Map` (and not for subtypes) should work best here: type information bubbles up through the type hierarchy, so that once `Map` of a parent type is detected and type "upgraded", subtypes will (when resolution call stack is unwound) also stay `MapLikeType`s.
This change went in 2.7; earlier code did indeed require check like what is used here.